docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+
+Once the import process is finished, create a `source` folder:
+```bash
+mkdir ~/code/uvisor-example/source
+```
+and place a new file `main.cpp` in it:
+
+```C
+/* ~/code/uvisor-example/source/main.cpp */
+
+#include "mbed.h"
+#include "rtos.h"
+
+DigitalOut led(LED1);
+
+int main(void)
+{
+    while (true) {
+        led = !led;
+        Thread::wait(500);
+    }
+}
+```
+
+This simple application just blinks an LED from the main thread, which is created by default by the OS.
+
+---
+
+**Checkpoint**
+
+Compile the application:
+
+```bash
+$ mbed compile -m K64F -t GCC_ARM
+```
+
+The resulting binary will be located at:
+
+```bash
+~/code/uvisor-example/.build/K64F/GCC_ARM/uvisor-example.bin
+```
+
+Drag-and-drop it onto the USB device mounted on your computer in order to flash the device. When the flashing process is completed, press the reset button on the device. You should see the device LED blinking.
+
+---
+
+In the next sections you will see:
+
+* How to [enable uVisor](#enable-uvisor) on the `uvisor-example` app.
+* How to [add a secure box](#add-a-secure-box) to the `uvisor-example` app with exclusive access to a timer, to a push-button interrupt, and to static and dynamic memories.
+
+## Enable uVisor
+[Go to top](#overview)
+
+To enable the uVisor on the app, just add the following lines at the beginning of the `main.cpp` file:
+
+```C
+/* ~/code/uvisor-example/source/main.cpp */
+
+#include "mbed.h"
+#include "rtos.h"
+#include "uvisor-lib/uvisor-lib.h"
+
+/* Register privleged system IRQ hooks.
+ * This is a system-wide configuration and it is independent from the app, but
+ * for the moment it needs to be specified in the app. This will change in a
+ * later version: The configuration will be provided by the OS. */
+extern "C" void SVC_Handler(void);
+extern "C" void PendSV_Handler(void);
+extern "C" void SysTick_Handler(void);
+UVISOR_SET_PRIV_SYS_IRQ_HOOKS(SVC_Handler, PendSV_Handler, SysTick_Handler);
+
+/* Main box ACLs. */
+/* Note: These are specific to the NXP FRDM-K64F board. See the section below
+ *       for more information. */
+static const UvisorBoxAclItem g_main_box_acls[] = {
+    /* For the LED */
+    {SIM,   sizeof(*SIM),   UVISOR_TACLDEF_PERIPH},
+    {PORTB, sizeof(*PORTB), UVISOR_TACLDEF_PERIPH},
+
+    /* For messages printed on the serial port. */
+    {OSC,   sizeof(*OSC),   UVISOR_TACLDEF_PERIPH},
+    {MCG,   sizeof(*MCG),   UVISOR_TACLDEF_PERIPH},
+    {UART0, sizeof(*UART0), UVISOR_TACLDEF_PERIPH},
+};
+
+/* Enable uVisor, using the ACLs we just created. */
+UVISOR_SET_MODE_ACL(UVISOR_ENABLED, g_main_box_acls);
+
+/* Rest of the existing app code */
+...
+```
+
+If you enable uVisor in the `blinky` app as it was written above, you will not get any particular security feature. All code and resources share the same security context, which we call the *main box*.
```

Comment:
I'm not sure when we want to start calling the _main box_ the _legacy box_. Maybe now?